src/MCPClient/lib/clientScripts/removeUnneededFiles.py

```diff
@@ -2,30 +2,41 @@
 
 import os
 import sys
+import ntpath
 
 # databaseFunctions requires Django to be set up
 import django
 django.setup()
 # archivematicaCommon
 from custom_handlers import get_script_logger
 from databaseFunctions import fileWasRemoved
+from main.models import File
 
 REMOVEABLE_FILES = ["Thumbs.db", "Icon", u"Icon\u000D"]
 
-def remove_file(target_file, file_uuid):
-    basename = os.path.basename(target_file)
-    if basename in REMOVEABLE_FILES:
-        print "Removing {filename} (UUID: {uuid})".format(uuid=file_uuid, filename=basename)
-        os.remove(target_file)
-        # Gearman passes parameters as strings, so None (NoneType) becomes
-        # "None" (string)
-        if file_uuid and file_uuid != "None":
-            fileWasRemoved(file_uuid)
+def remove_file(sip_obj_dir, sip_UUID):
+
+    ### 1. Remove unneeded files from package.
+    for dir_path, dir_name, filenames in os.walk(sip_obj_dir):
+        for remove_name in REMOVEABLE_FILES:
+            for filename in filenames:
+                if remove_name == filename:
+                    del_file = dir_path + '/' + filename
+                    os.remove(del_file)
+                    logger.debug('removeUnneededFiles file deleted: %s', del_file)
+
+    ### 2. Insert removal notification into events database
+    sip_files = File.objects.filter(sip=sip_UUID).values_list('uuid', 'currentlocation')
```

Comment:
While I think it's slightly less efficient, I think it makes for clearer and less error-prone code if the `fileWasRemoved` call happens in the same loop as the file being deleted, so the database is kept as close to in sync as the disk as possible. I'm open to discussion on this, especially if it is much less efficient, so please tell me if you disagree.